tools/devops/automation/templates/main-stage.yml

```diff
@@ -73,28 +73,123 @@ parameters:
 - name: macOSName
   type: string
 
-- name: simTestsConfigurations
+# Ideally we should read/get the list of platforms from somewhere else, instead of hardcoding them here.
+# Note that this is _all_ the platforms we support (not just the enabled ones).
+- name: supportedPlatforms
   type: object
   default: [
-    'bcl',
-    'cecil',
-    'dotnettests',
-    'fsharp',
-    'framework',
-    'generator',
-    'interdependent-binding-projects',
-    'install-source',
-    'introspection',
-    'linker',
-    'mac-binding-project',
-    'mmp',
-    'mononative',
-    'monotouch',
-    'msbuild',
-    'mtouch',
-    'xammac',
-    'xcframework',
-    'xtro'
+    {
+      platform: iOS,
+      isDotNetPlatform: true,
+      isLegacyPlatform: true
+    },
+    {
+      platform: macOS,
+      isDotNetPlatform: true,
+      isLegacyPlatform: true
+    },
+    {
+      platform: tvOS,
+      isDotNetPlatform: true,
+      isLegacyPlatform: true
+    },
+    {
+      platform: watchOS,
+      isDotNetPlatform: false,
+      isLegacyPlatform: true
+    },
+    {
+      platform: MacCatalyst,
+      isDotNetPlatform: true,
+      isLegacyPlatform: false
+    },
+    {
+      # when running platform-specific test runs, we also need a special test run that executes tests that only runs when multiple platforms are enabled
```

Comment:
Off the top of my head these two tests should be categorized to run for the multiple pass (since they require multiple platforms enabled to be executed):

https://github.com/xamarin/xamarin-macios/blob/1bdb8d64e6915273e4e168bedeb4f8659b832685/tests/dotnet/UnitTests/PackTest.cs#L105-L111

https://github.com/xamarin/xamarin-macios/blob/1bdb8d64e6915273e4e168bedeb4f8659b832685/tests/dotnet/UnitTests/ProjectTest.cs#L1040-L1044

However, I might be forgetting something. It would be nice if you could collect the platform-specific test results from this PR, merge them all, and compare with a test run from main to see if there are any tests we're missing.

Reply:
Still missing action in this thread :)

Reply:
Did you compare the test results from the dotnettest platform-specific+multple test runs in this PR with a test run from main to make sure we're not skipping any tests in this PR?

Reply:
Still working out how to do that. I have verified that the Multiple test run is in fact running the correct set of tests.

Reply:
Ok, So we are missing the following tests:
```
> BindingXcFrameworksProject(iOS,False)
> BindingXcFrameworksProject(iOS,True)
> BindingXcFrameworksProject(MacCatalyst,False)
> BindingXcFrameworksProject(MacCatalyst,True)
> BindingXcFrameworksProject(MacOSX,False)
> BindingXcFrameworksProject(MacOSX,True)
> BindingXcFrameworksProject(TVOS,False)
> BindingXcFrameworksProject(TVOS,True)
```
And we are running `AreAllTemplatesListed` 3 more times than previously

Reply:
> And we are running `AreAllTemplatesListed` 3 more times than previously

I think that's fine, it's a fast test that it's not worth spending the time to change.